ortho_seq_code/plotclass.py

```diff
@@ -0,0 +1,361 @@
+            )
+            return
+        if type(list) == list:
+            if len(span) > 2 or not np.all(
+                [type(i) == float or type(i) == int for i in self.x_flat]
+            ):
+                print("Span must be a 2d list of ints or floats.")
+                return
+        x_flat = np.array(self.x_flat)
+        x = self.x
+        if comp == "<":
+            if type(span) == list:
+                x[x < span[0]] = 0
+                x_flat[x_flat < span[0]] = 0
+            else:
+                x[x < span] = 0
+                x_flat[x_flat < span] = 0
+        elif comp == ">":
+            if type(span) == list:
+                x[x > span[1]] = 0
+                x_flat[x_flat > span[1]] = 0
+            else:
+                x[x > span] = 0
+                x_flat[x_flat > span] = 0
+        elif comp == "<>":
+            if type(span) != list:
+                print("<> and >< can only be used for list ranges.")
+                return
+            x[abs(x - np.mean(span)) >= np.mean(span)] = 0
+            x_flat[abs(x_flat - np.mean(span)) >= np.mean(span)] = 0
+        else:
+            if type(span) != list:
+                print("<> and >< can only be used for list ranges.")
+                return
+            x[abs(x - np.mean(span)) < np.mean(span)] = 0
+            x_flat[abs(x_flat - np.mean(span)) < np.mean(span)] = 0
+        print("Successfully trimmed array.")
+        self.x_flat = list(x_flat)
+        self.x = x
+
+    def histogram(
+        self,
+        bins=None,
+        site=None,
+        alphabet_item=None,
+        omit_zeroes=True,
+        bin_color=None,
+        border=True,
+        out_dir=None,
+    ):
+        if alphabet_item is not None and site is not None:
+            print("Only one of site and alphabet_item may not be null.")
+            return
+        if alphabet_item is not None and alphabet_item not in self.alphbt_input:
+            print("Alphabet item must be one of:")
+            print(self.alphbt_input)
+            return
+        if site is not None:
+            if site > self.s or site < 0:
+                print("Site must be between 0 and", str(self.s))
+                return
+        if alphabet_item is not None:
+            a = self.alphbt_input.index(alphabet_item) * self.s
+            x_red = self.x_flat[a : a + self.d]
+        elif site is not None:
+            x_red = self.x[site]
+        else:
+            x_red = self.x_flat
+        if omit_zeroes:
+            x_red = np.array(x_red)[np.array(x_red) != 0].tolist()
+        if border:
+            if bins is not None:
+                width = 1 - bins / 200
+            else:
+                width = 1 - len(x_red) / 1200
+        else:
+            width = 0
+        plt.hist(x_red, bins=bins, color=bin_color, lw=width, ec="black")
+        plt.ylabel(self.phenotype)
+        if out_dir is not None:
+            path_sav = "rFon1D_hist_" + str(self.phenotype) or "" + ".png"
```

Comment:
can this path_sav be in the init function? so you are not repeating in every plot function?

Reply:
The name varies depending on what you're doing. For example, it's "rFon1D_hist_" for histograms, but "rFon1D_heatmap_" for heatmaps.

Reply:
Got it!